### ISSUE TITLE
Include stuff from StrongholdPlayer?

### ISSUE THREAD

**Issue by ByBurton** [Sh0wdown/UnofficialCrusaderPatch#473]
I've taken quite a liking to [this mod](https://www.moddb.com/mods/shcplayer). Problem is, it does not work with the version of Crusader required by the UCP. 
Is there a way to import / include the stuff fixed in there? 
I tried contacting the guy who coded it but with no luck so far.

There are such awesome features in it. 
- Recording and playing matches?
- Viewing the AIV castles ingame
- Custom hotkeys
- Fixed a lot of bugs

**Comment by GRhin** [comment]
I have spoken with Juggernaunt, he doesnt feel like its worth his time to update his mode to work with other versions of the game, and even if he did, from my own testing, it still wouldnt work with the UCP, it would be one or the other.
The recording and playing matches and the hotkeys are my favorite features of that mod, if he made it usable on other versions I would use that for multiplayer gaming over UCP, but id still use UCP for the AI stuff.

**Comment by ByBurton** [comment]
@GRhin that's that he told me too. Only thing he ever replied to me. 
But maybe @Sh0wdown can reverse engineer some stuff from his mod. I don't know.
So many cool bugfixes in there.

**Comment by J-T-de** [comment]
Yeah, I think at least the bugfixes should be included, as it really is within the scope of an unofficial community patch. The hotkeys should not be too hard, but the other two plugins seem to be rather large.

Unfortunately the source code of SHCPlayer is not available. Maybe @JuGGerNaunT is interested in sharing some parts of his awsome work, maybe this would make it easier for Showdown to include some parts.

Also I want to point out, that JuGGerNaut was able to add additional items into the menu, which is I think a really interesting hidden feature, as it might be useful for some content creators, like additional campaigns... Something like this idea https://github.com/Sh0wdown/UnofficialCrusaderPatch/issues/345 or idk?

@ByBurton : The link is (at least for me) dead due to the part after the &, maybe you want to edit that.

**Comment by GRhin** [comment]
He isnt willing to share his code at this stage. He has future plans that
he might get to, and doesn't want to release code until hes decided he wont
follow those plans.

**Comment by Krarilotus** [comment]
I think i know how the castle viewer works kinda
If you save a game with AI on a map in a certain state, you can then transform that save into a map , load it into the editor, and the ai will just place all its buildings down in fast speed.
When i can reproduce that 100%, i will share it with you.
But it could be, that he just makes the AI build their castle in editor mode, just having 0 set for buildstep time.

You should try the mod with ironmines placed in the aiv, if it does not build those without iron patches on the preview map, it might be exactly his method.
Else i dont know :P
